resources/assets/components/Signup/index.js

```diff
@@ -321,67 +350,74 @@ class Signup extends React.Component {
     const posts = this.state.posts;
 
     return (
-      <div className="signup">
-        <div className="container__row">
-          <div className="container__block -half">
-            <UserInformation user={user}>
-              <TextBlock title="Why Statement" content={signup.why_participated} />
-            </UserInformation>
-          </div>
-
-          <div className="container__block -half">
-            <Quantity quantity={signup.quantity} noun={campaign.reportback_info.noun} verb={campaign.reportback_info.verb} />
-
-            <div className="container__row">
-
-              <a href="#" onClick={e => this.showHistory(signup.signup_id, e)}>Edit | Show History</a>
-
-              <ModalContainer>
-                {this.state.displayHistoryModal ?
-                  <HistoryModal
-                    id={this.state.historyModalId}
-                    onUpdate={this.updateQuantity}
-                    onClose={e => this.hideHistory(e)}
-                    signup={signup}
-                    campaign={campaign}
-                    signupEvents={this.state.signupEvents}
-                  />
-                  : null}
-              </ModalContainer>
+      this.state.signup != undefined ?
```

Comment:
It might make sense to have this as a guard clause instead (since there's so much in this ternary):

```js
if (! this.state.signup) {
  return <Empty header={`This signup has been deleted`} />;
}

return (
  <div className="signup">
    <div className="container__row">
        // ...

```